### PR TITLE
improve error message

### PR DESCRIPTION
**Pull request by tpolecat**
Added the path and type and a suggestion that a `LeafMapping` is probably missing in the case where there's no encoder available for a result value.

Before:

```
Not a leaf: 0
```

After:

```
Cannot encode value 0 at programs/plannedTime/pi/microseconds (of GraphQL type NonNegLong). Did you forget a LeafMapping?
```

### CODE REVIEW

File: modules/sql/src/main/scala/SqlMapping.scala [hunk 1]
```diff
@@ -3064,7 +3064,9 @@ trait SqlMapping[F[_]] extends CirceMapping[F] with SqlModule[F] { self =>
     def isLeaf: Boolean = tpe.isLeaf
 
     def asLeaf: Result[Json] =
-      mapped.encoderForLeaf(tpe).map(enc => enc(focus).rightIor).getOrElse(mkErrorResult(s"Not a leaf: $focus"))
+      mapped.encoderForLeaf(tpe).map(enc => enc(focus).rightIor).getOrElse(mkErrorResult(
+        s"Cannot encode value $focus at ${context.path.reverse.mkString("/")} (of type of type ${context.tpe}). Did you forget a LeafMapping?".stripMargin.trim
```
Comment (swalker2m) [inline modules/sql/src/main/scala/SqlMapping.scala:3068]: `of type of type`

**Review by swalker2m — COMMENTED** [review]
Excellent, newbies like me would benefit greatly from this kind of thing.

File: modules/sql/src/main/scala/SqlMapping.scala (hunk 1 above)
Comment (tpolecat) [reply modules/sql/src/main/scala/SqlMapping.scala:3068]: doh

Comment (tpolecat) [reply modules/sql/src/main/scala/SqlMapping.scala]: Fixed (description above edited).

**Review by milessabin — APPROVED** [review]
LGTM!

**Pull request merged**